server/app-engine/static/service-alert-ui.js

```diff
@@ -0,0 +1,612 @@
+  util.populateList("route-select", "Select a Route ID", getItems("routes", "route_id"));
+  util.populateList("trip-select", "Select a Trip ID", getItems("trips", "trip_id"));
+  util.populateList("stop-select", "Select a Stop ID", getItems("stops", "stop_id"));
+}
+
+function getItems(type, columnName){
+  return fileMap.get(type).map(function (el) { return el[columnName]; })
+}
+
+function handleEntitySelection(checkbox){
+  util.log("handleEntitySelection()");
+  let dropdownName = checkbox.id.slice(0,-8) + "select";
+  if(checkbox.checked){
+    util.showElement(dropdownName);
+  } else {
+    util.hideElement(dropdownName);
+    util.resetDropdownSelection(dropdownName);
+  }
+}
+
+function getValue(id){
+  let value = document.getElementById(id).value;
+  if(value === "disabled" || value === '') return null;
+  else return value;
+}
+
+async function postServiceAlert() {
+    util.log('handleGPSUpdate()');
+    let timestamp = Math.floor(Date.now() / 1000);
+    let cause = getValue("cause-select");
+    let effect = getValue("effect-select");
+    let description = getValue("description");
+    let header = getValue("header");
+    let startTime = getValue("start-time");
+    let stopTime = getValue("stop-time");
+    let startTimeSecs = new Date(startTime).getTime() / 1000;
+    let stopTimeSecs = new Date(stopTime).getTime() / 1000;
+    let agency_id = getValue("agency-select");
+    let route_id = getValue("route-select");
+    let trip_id = getValue("trip-select");
+    let route_type = getValue("route-type-select");
+    let stop_id = getValue("stop-select");
+    let url = getValue("url");
+
+    if(agency_id === null && route_id === null && stop_id === null && trip_id === null && route_type === null){
+      alert("Please select at least one entity for your alert");
+      return;
+    }
+
+    if(cause === null || effect === null){
+      alert("Please select both a cause and an effect for your alert");
+      return;
+    }
+
+    if(header === null || description === null){
+      alert("Please write both a header and a description");
+      return;
+    }
+
+    if(startTime < 0 || stopTime < 0){
+      alert("Please select start and stop dates that are in the future");
+      return;
+    }
+
+    if(stopTime < startTime){
+      alert("Start time must be before stop time");
+      return;
+    }
+
+    let data = {
+      agency_key: agencyID,
+      timestamp: timestamp,
+      time_start: startTimeSecs,
+      time_stop: stopTimeSecs,
+      agency_id: agency_id,
+      trip_id: trip_id,
+      stop_id: stop_id,
+      route_id: route_id,
+      route_type: route_type,
+      cause: cause,
+      effect: effect,
+      header: header,
+      description: description
+    };
+
+    if(url !== null){
+      urlData = {url: url}
+      Object.assign(data, urlData)
+    }
+
+    util.signAndPost(data, signatureKey, '/post-alert', document);
+
+    // Consider actually confirming send status
```

Comment:
Are you saying it's unclear if post succeeded? If so, then we shouldn't proclaim success.

Reply:
Yes, it's not good, I need to update this